02_housing/Homelessness_metrics_city.do

```diff
@@ -1,15 +1,32 @@
 ** HOMELESSNESS **
 ** E Blom **
 ** 2020/08/04 **
-*Updated September 2022 by Emily Gutierrez
-*Updated February 2024 by Emily  Gutierrez
-*Creates the number and share of homeless students by city for 2020-21 & 2021-22 (2014-15 through 2019-20 completed in previous update)
-	*Creates the number and share of homeless by race/ethnicity for 2020-21 & 2021-22 (recreates 2019-20 with total homeless as denominator)
-	
+	*Original code from E Blom
+	** Data suppression: data are suppressed when values are between 0-2, but if only one value is suppressed the next smallest number is also suppressed ** 
+	** Original code replaced all suppressed data with the midpoint (1) but this does not yield numbers (for 2017) that align perfectly with this report: 
+	** https://nche.ed.gov/wp-content/uploads/2020/01/Federal-Data-Summary-SY-15.16-to-17.18-Published-1.30.2020.pdf (Tables 5 and 6)
+	** Note also that data are unduplicated * by LEA * which does not mean they will necessarily be unduplicated * by county * if students switch between LEAs in a county **
+*Updated September 2022 by E Gutierrez
+*Updated February 2024 by E Gutierrez
+	*Creates the number and share of homeless students by city for 2020-21 & 2021-22 (2014-15 through 2019-20 completed in previous update)
+		*Creates the number and share of homeless by race/ethnicity for 2020-21 & 2021-22 (recreates 2019-20 with total homeless as denominator)
+			/*Raw data is now posted on EdDataExpress instead of EdFacts. 
+			The data posted on EdDataExpress does not include the subgrant_status variable. 
+			According to EdFacts documentation, the variable is used to determine suppression information through 2018-19, 
+			and is therefore used in our own suppression determinations for data up through 2018-19. Starting in 2019-20, 
+			subgrant_status is no longer used to determine suppression information, and is therefore not necessary for this or future updates.*/
+*Updated December 2024 by E Gutierrez
+	*Creates the total number and share of homeless students by city for 2014-15 through 2018-19 (2014-2018)
+
+**Housekeeping: install educationdata command **
+cap n ssc install libjson
+net install educationdata, replace from("https://urbaninstitute.github.io/education-data-package-stata/")
+
+*Set up globals and directories
 clear all
 
-global gitfolder "C:\Users\ekgut\OneDrive\Desktop\urban\Github2\mobility-from-poverty"
-global years 2019 2020 2021 // refers to 2019-20 school year through most recent data
+global gitfolder "C:\Users\ekgut\OneDrive\Desktop\urban\Github\mobility-from-poverty"
+global years 2014 2015 2016 2017 2018 
```

Comment:
For someone who isn't familiar with school data it would be helpful to comment or document somewhere whether the year refers to the starting or ending year of the school year (I think it's ending but not 100% certain). I have seen issues with split year data like this before where two programmers weren't on the same page.